From b09d1e1e6ff7d2c79fa1e442265db68b8a747e2b Mon Sep 17 00:00:00 2001
From: Byron McCollum <byron.mccollum@rackspace.com>
Date: Thu, 05 Nov 2015 20:07:23 -0600
Subject: [PATCH] Add retry and delay to RabbitMQ queue mirroring setup.

Occasionally, RabbitMQ will not be fully up before trying to setup queue
mirroring. This change adds a retry with delay to the task to help coax
it into being successful.

Change-Id: Ifee971176dc9649131778f7fe2fe16144509a1b1
Closes-bug: #1513668
---

diff --git a/playbooks/roles/rabbitmq_server/tasks/rabbitmq_post_install.yml b/playbooks/roles/rabbitmq_server/tasks/rabbitmq_post_install.yml
index e620331..5a40e0f 100644
--- a/playbooks/roles/rabbitmq_server/tasks/rabbitmq_post_install.yml
+++ b/playbooks/roles/rabbitmq_server/tasks/rabbitmq_post_install.yml
@@ -28,6 +28,9 @@
     pattern: '^(?!amq\.).*'
     tags: "ha-mode=all"
   register: rabbitmq_queue_mirror
+  until: rabbitmq_queue_mirror | success
+  retries: 3
+  delay: 30
   tags:
     - rabbitmq-config
     - rabbitmq-cluster
